packages/zowe-explorer/src/trees/uss/UssFSProvider.ts

```diff
@@ -429,6 +430,26 @@ export class UssFSProvider extends BaseProvider implements vscode.FileSystemProv
      */
     public async readFile(uri: vscode.Uri): Promise<Uint8Array> {
         let file: UssFile | UssDirectory;
+
+        // Check if the profile for URI is not zosmf, if it is not, create a deferred promise for the profile.
+        // If the extenderTypeReady map does not contain the profile, create a deferred promise for the profile.
+        const profileName = uri.path.split("/")[1];
+        const profileInfo = new ProfileInfo("zowe");
+        await profileInfo.readProfilesFromDisk();
+
+        if (
+            profileInfo.getAllProfiles("zosmf").filter((profile) => profile.profName === profileName).length <= 0 &&
+            profileInfo.getTeamConfig().api.profiles.exists(profileName) &&
```

Comment:
We'll want to avoid reading profiles as part of reading a file in the filesystem. Is there a way that we can access the existing instance, using something like `Profiles.getInstance().allProfiles` instead?